library/aesni.h

```diff
@@ -35,13 +35,13 @@
 /* Can we do AESNI with inline assembly?
  * (Only implemented with gas syntax, only for 64-bit.)
  */
-#if defined(MBEDTLS_HAVE_ASM) && defined(__GNUC__) && \
```

Comment:
We seem to have lost the check for `__GNUC__` - is this intentional? Will this work on compilers other than clang/gcc?

Reply:
Intrinsic code supports MSVC also.

Reply:
Beside that, I think this check should be global. That's for checking target platform, I do not think it should be in modules